src/components/dialogs/inputs/reactive-capability-curve-table.js

```diff
@@ -0,0 +1,212 @@
+
+const minimalValue = { p: '', qminP: '', qmaxP: '' };
+
+function getId(value) {
+    return value?.id ? value.id : 'defaultId';
+}
+
+function getNewId() {
+    return Math.random().toString(36).substring(2);
+}
+
+function hasValueCorrectFormat(valueToTest, index) {
+    return !!(
+        valueToTest &&
+        valueToTest[index] &&
+        valueToTest[index].p !== undefined &&
+        valueToTest[index].qminP !== undefined &&
+        valueToTest[index].qmaxP !== undefined
+    );
+}
+
+function enforceMinimumValues(valueToTest) {
+    let returnValue = [];
+    returnValue[0] = hasValueCorrectFormat(valueToTest, 0)
+        ? valueToTest[0]
+        : minimalValue;
+    returnValue[1] = hasValueCorrectFormat(valueToTest, 1)
+        ? valueToTest[1]
+        : minimalValue;
+    if (valueToTest?.length > 2) {
+        for (let i = 2; i < valueToTest.length; i++) {
+            returnValue[i] = hasValueCorrectFormat(valueToTest, i)
+                ? valueToTest[i]
+                : minimalValue;
+        }
+    }
+    return returnValue;
+}
+
+export const useReactiveCapabilityCurveTableValues = ({
+    tableHeadersIds,
+    Field,
+    inputForm,
+    defaultValues, // format : either undefined or [{ p: '', qminP: '', qmaxP: '' }, { p: '', qminP: '', qmaxP: '' }]
+    isReactiveCapabilityCurveOn,
+    disabled = false,
+}) => {
+    const classes = useStyles();
+
+    // Values sent back to the parent component.
+    const [values, setValues] = useState(enforceMinimumValues(defaultValues));
+
+    // Values displayed in the form. When updated, they also updates "values".
+    const [displayedValues, setDisplayedValues] = useState(
+        enforceMinimumValues(defaultValues)
+    );
+
+    useEffect(() => {
+        // Updates the component when the correct default values are given by the parent component.
+        if (defaultValues !== undefined && defaultValues.length > 0) {
+            const enforcedMinimumDefaultValues =
+                enforceMinimumValues(defaultValues);
+            setValues(enforcedMinimumDefaultValues);
+            setDisplayedValues(enforcedMinimumDefaultValues);
+        }
+    }, [defaultValues]);
+
+    const handleDeleteItem = useCallback(
+        (index) => {
+            const newValues = [
+                ...values.slice(0, index),
+                ...values.slice(index + 1),
+            ];
+
+            setValues(newValues);
+            setDisplayedValues(newValues);
+
+            inputForm.reset();
+        },
+        [values, inputForm]
+    );
+
+    const handleAddValue = useCallback(() => {
+        const newPosition = values.length - 1;
+        const newValues = [
+            ...values.slice(0, newPosition),
+            minimalValue,
+            ...values.slice(newPosition),
+        ];
+
+        setValues(newValues);
+
+        // Adds a unique ID on each displayed line, to prevent a bad rendering effect
+        let newValuesForDisplay = [];
```

Comment:
Could be written with Array.prototype.map
```
const newValuesForDisplay = newValues.map((v) => {
  v.id = getNewId();
  return v;
});
```

Reply:
Yes, thank you.